lib/rabbitmq/http/client.rb

```diff
@@ -129,8 +129,13 @@ def declare_exchange(vhost, name, attributes = {})
         decode_resource(response)
       end
 
-      def delete_exchange(vhost, name)
-        decode_resource(@connection.delete("exchanges/#{uri_encode(vhost)}/#{uri_encode(name)}"))
+      def delete_exchange(vhost, name, if_unused = false)
+        query = if if_unused
+                  "?if-unused=true"
```

Comment:
I think the Faraday way is to [set query parameters by mutating a hash](https://github.com/lostisland/faraday).